### PR TITLE
Scanner's recognition of restricted identifier "when" is fragile

### PR DESCRIPTION
**Pull request by srikanth-sankaran**
## Author checklist

- [ ] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by iloveeclipse** [comment]
```
15:18:26  3 API ERRORS
15:18:26  [API ERROR] File MANIFEST.MF at line 5: The major version should be incremented in version 3.36.0, since API breakage occurred since version 3.35.0 - The super interfaces set has been reduced (java.lang.constant.Constable) for type org.eclipse.jdt.core.util.ExternalAnnotationUtil.MergeStrategy
15:18:26  - The super interfaces set has been reduced (java.lang.constant.Constable) for type org.eclipse.jdt.core.dom.rewrite.ImportRewrite.TypeLocation (location: /home/jenkins/agent/workspace/eclipse.jdt.core-Github_PR-1436/org.eclipse.jdt.core/META-INF/MANIFEST.MF)
15:18:26  [API ERROR] File ImportRewrite.java at line 116: The super interfaces set has been reduced (java.lang.constant.Constable) for type org.eclipse.jdt.core.dom.rewrite.ImportRewrite.TypeLocation (location: /home/jenkins/agent/workspace/eclipse.jdt.core-Github_PR-1436/org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/rewrite/ImportRewrite.java)
15:18:26  [API ERROR] File ExternalAnnotationUtil.java at line 68: The super interfaces set has been reduced (java.lang.constant.Constable) for type org.eclipse.jdt.core.util.ExternalAnnotationUtil.MergeStrategy (location: /home/jenkins/agent/workspace/eclipse.jdt.core-Github_PR-1436/org.eclipse.jdt.core/model/org/eclipse/jdt/core/util/ExternalAnnotationUtil.java)
```

Looks like PDE bug...

**Comment by iloveeclipse** [comment]
It looks like if the API baseline would see java.lang.Enum from Java 8 or 11 in the **current** API (why???) and compare it with java.lang.Enum from Java 17 **in the API baseline** where it **additionally** implements Constable (since Java 12).

So it would be as if PDE is not able to recognize Java 21 as valid JDK and fails back to Java 11 (or 8). Something probably related to https://github.com/eclipse-pde/eclipse.pde/issues/764 where test complains that `Classpath entry for JavaSE-17 is missing`.

**Comment by iloveeclipse** [comment]
@srikanth-sankaran :  one just should keep trying / rebuilding - at some point in time API tooling gives up...
See https://github.com/eclipse-pde/eclipse.pde/issues/771

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/formatter/FormatterRegressionTests.java [hunk 1]
```diff
@@ -16223,4 +16223,46 @@ public void testBug564052() {
...
+
+		    void foo() {
+		            when(myObj);
+		        foo(myObj);
+		        foo(myObj);
+		    }
+
+		    void foo( Object stuff ) {}
+		    void when( Object methodCall ) {}
+		}
```
Comment (stephan-herrmann) [inline org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/formatter/FormatterRegressionTests.java:16246]: It seems here you introduced a bogus `\r\n` lineend, which then escalated in https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1566/files#r1395990810 (_github takes long time to actually show that comment due to size of the change_).

Reminds me that mixed line endings are dangerous :)

**Pull request merged**